homeassistant/components/utility_meter/sensor.py

```diff
@@ -655,6 +663,19 @@ def async_source_tracking(event):
 
         self.async_on_remove(async_at_started(self.hass, async_source_tracking))
 
+        async def async_track_time_zone(event):
+            """Reconfigure Scheduler after time zone changes."""
+
+            if self._current_tz != self.hass.config.time_zone:
+                self._current_tz = self.hass.config.time_zone
+
+                self._config_scheduler()
+                await self._program_reset()
```

Comment:
Side note: `_program_reset` doesn't need to be a coroutine function since there's no await inside that function.